content/learn/book/migration-guides/0.9-0.10/_index.md

```diff
@@ -0,0 +1,733 @@
+where T: ExclusiveSystemParamFunction<Param>
+{ ... }
+
+// After
+fn my_generic_system<T, In, Out, Param>(system_function: T)
+where T: ExclusiveSystemParamFunction<In, Out, Param>
+{ ... }
+```
+
+### [Document alignment requirements of `Ptr`, `PtrMut` and `OwningPtr`](https://github.com/bevyengine/bevy/pull/7151)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+- Safety invariants on `bevy_ptr` types’ `new` `byte_add` and `byte_offset` methods have been changed. All callers should re-audit for soundness.
+
+### [Panic on dropping NonSend in non-origin thread.](https://github.com/bevyengine/bevy/pull/6534)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+Normal resources and `NonSend` resources no longer share the same backing storage. If `R: Resource`, then `NonSend<R>` and `Res<R>` will return different instances from each other. If you are using both `Res<T>` and `NonSend<T>` (or their mutable variants), to fetch the same resources, it’s strongly advised to use `Res<T>`.
+
+### [Remove the `SystemParamState` trait and remove types like `ResState`](https://github.com/bevyengine/bevy/pull/6919)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+Note: this replaces the migration guide for #6865.
+This is relative to Bevy 0.9, not main.
+
+The traits `SystemParamState` and `SystemParamFetch` have been removed, and their functionality has been transferred to `SystemParam`.
+
+```rust
+// Before (0.9)
+impl SystemParam for MyParam<'_, '_> {
+    type State = MyParamState;
+}
+unsafe impl SystemParamState for MyParamState {
+    fn init(world: &mut World, system_meta: &mut SystemMeta) -> Self { ... }
+}
+unsafe impl<'w, 's> SystemParamFetch<'w, 's> for MyParamState {
+    type Item = MyParam<'w, 's>;
+    fn get_param(&mut self, ...) -> Self::Item;
+}
+unsafe impl ReadOnlySystemParamFetch for MyParamState { }
+
+// After (0.10)
+unsafe impl SystemParam for MyParam<'_, '_> {
+    type State = MyParamState;
+    type Item<'w, 's> = MyParam<'w, 's>;
+    fn init_state(world: &mut World, system_meta: &mut SystemMeta) -> Self::State { ... }
+    fn get_param<'w, 's>(state: &mut Self::State, ...) -> Self::Item<'w, 's>;
+}
+unsafe impl ReadOnlySystemParam for MyParam<'_, '_> { }
+```
+
+The trait `ReadOnlySystemParamFetch` has been replaced with `ReadOnlySystemParam`.
+
+```rust
+// Before
+unsafe impl ReadOnlySystemParamFetch for MyParamState {}
+
+// After
+unsafe impl ReadOnlySystemParam for MyParam<'_, '_> {}
+```
+
+### [Extend EntityLocation with TableId and TableRow](https://github.com/bevyengine/bevy/pull/6681)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+A `World` can only hold a maximum of 232 - 1 archetypes and tables now. If your use case requires more than this, please file an issue explaining your use case.

```

Comment:
```suggestion
A `World` can only hold a maximum of 2<sup>32</sup> - 1 archetypes and tables now. If your use case requires more than this, please file an issue explaining your use case.
```

Not sure why the superscript here didn't transfer over.

Reply:
Right now this is generate by parsing the section as markdown and then rewriting markdown manually not just pasting the code directly. It's worked fine before, but I'm not sure why it skipped this